guacamole-common-js/src/main/webapp/modules/AudioRecorder.js

```diff
@@ -183,11 +183,12 @@ Guacamole.RawAudioRecorder = function RawAudioRecorder(stream, mimetype) {
 
     /**
      * A function which directly invokes the browser's implementation of
-     * navigator.getUserMedia() with all provided parameters.
+     * navigator.mediaDevices.getUserMedia() with all provided parameters.
      *
      * @type Function
      */
-    var getUserMedia = (navigator.getUserMedia
+    var getUserMedia = (navigator.mediaDevices.getUserMedia
```

Comment:
I'm not sure this will work:

1. If `navigator.mediaDevices` is not defined (older browsers), then the fallback to `navigator.getUserMedia`, etc. will not take effect as the attempt to access `navigator.mediaDevices.getUserMedia` will fail. The example at https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia gets around this by ensuring `navigator.mediaDevices` is set to `{}` if not otherwise defined, but I'm sure there are other and/or better approaches.
2. The call to `(...).bind(navigator)` creates a function which automatically uses `navigator` for `this`. This is correct for functions like `navigator.getUserMedia()`, but not for `navigator.mediaDevices.getUserMedia()`. I have not checked, but I expect that `navigator.mediaDevices.getUserMedia()` would need `this` to be `navigator.mediaDevices`.

Rather than point directly to `getUserMedia()`, we could instead point to `mediaDevices`, invoking the function later on as `mediaDevices.getUserMedia()`, defining a `mediaDevices` object with a `getUserMedia` property if not already provided on `navigator`. That might solve things cleanly, and the `bind()` would only be necessary in the `navigator` fallback case.

Reply:
In fact ... it might be even better to simply:

1. Test whether `mediaDevices` is defined on `navigator`.
2. If not, define a shim `mediaDevices` object on `navigator` having a single `getUserMedia` property pointing to the function we used to reference here.
3. Use `navigator.mediaDevices.getUserMedia()` directly where we previously used the locally-defined `getUserMedia()`.

Reply:
Okay...I'll take a stab at that...